htdocs/js/datepicker-fixes.js

```diff
@@ -0,0 +1,60 @@
+$(document).ready(function () {
+    //ncote
+    //always use the jquery datepicker so we can control the format
+    //htdocs/js/polyfills.js does some of this when native datepicker is not available so might cause a conflict on older browsers
+    //  esp. the event listener for keydown, which ignores keypresses
+    var inputDateFormat = 'dd-M-yy';
+    var inputDateformatPlaceholder = 'dd-MMM-yyyy';  //user-friendly format hint
+    var serverDateFormat = 'yy-mm-dd';
+
+    var dateInputs = $('input[type=date]');
+
+    //disable native datepicker, i.e. turn them into normal text inputs
+    dateInputs.attr('type', 'text');
+    dateInputs.attr('placeholder', inputDateformatPlaceholder);
+
+    //to safely load and save date values, use safe format yyyy-mm-dd in an alternate hidden field
+    //rename existing date fields to ...-visible, and inject fields with original names to hold the universal format
+    dateInputs.each(function (index, element) {
+        var $element = $(element);
+        var name = $element.attr('name');
+        $element.attr('name', '');  //blank name will prevent submission
+        $element.attr('id', name + '_visible_date_field');
+        var $backingElement = $('<input type="hidden" name="' + name + '">');  //give original name to our locale-indepdent date field
+        $backingElement.val($element.val());
+        $element.after($backingElement);
+
+        $element.datepicker({
+            dateFormat: inputDateFormat,  //ncote: CRU standard e.g. 27-Feb-2018, but make configurable (in Smarty templates and React components as well)
```

Comment:
@ridz1208 do you know the context/motivation behind this code? i'm trying to get an idea of what the goal of this fix is

Reply:
it appears to me that what we're doing here is forcing all input format to be dd/mm/yyyy. this is already the case for chrome, but it is true that on firefox, the date input is formatted as yyyy-mm-dd. merging this file will affect all date input elements however (the var inputDateFormat is not actually configurable as stated in the above comment) so do we want to make all dates dd/mm/yyyy on core? this is also done in an ..interesting way. i'm not sure why we are adding a hidden field and then copying its value into the visible field. it all seems a little weird to me. i'll look into finding a simpler way of making firefox input fields be mm-dd-yyyy if this is entirely what this file is intending to do

Reply:
In an ideal world, the format of all dates across loris would be a configuration option in loris.

For right now, if there is a simpler cleaner way of doing what this is trying to do, it would be nice to incorporate it to the CBIGR VM instead of porting this solution over